### PR TITLE
Reference Recipe Add Target Options

### PR DESCRIPTION
**Pull request by peterm790**
This is following from https://github.com/pangeo-forge/staged-recipes/pull/154#issuecomment-1192794733 

I think this should be sufficient to get the LiveOcean [recipe](https://github.com/pangeo-forge/staged-recipes/pull/154) to run. I have not set up any tests for this as I'm not entirely sure how that could be done.

### CODE REVIEW

File: pangeo_forge_recipes/recipes/reference_hdf_zarr.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def finalize(config: HDFReferenceRecipe):
             files,
             remote_protocol=remote_protocol,
             remote_options=config.netcdf_storage_options,
+            target_options=config.reference_storage_options,
```
Comment (rabernat) [inline pangeo_forge_recipes/recipes/reference_hdf_zarr.py:55]:
```suggestion
            target_options=config.target_options,
```

File: pangeo_forge_recipes/recipes/reference_hdf_zarr.py [hunk 2]
```diff
@@ -157,6 +158,7 @@ class HDFReferenceRecipe(BaseRecipe, StorageMixin, FilePatternMixin):
     output_json_fname: str = "reference.json"
     output_intake_yaml_fname: str = "reference.yaml"
     netcdf_storage_options: dict = field(default_factory=dict)
+    reference_storage_options: Optional[dict] = field(default_factory=dict)
```
Comment (rabernat) [inline pangeo_forge_recipes/recipes/reference_hdf_zarr.py:161]:
```suggestion
    target_options: Optional[dict] = field(default_factory=dict)
```

**Review by rabernat — COMMENTED** [review]
Thanks so much for this contribution!

It's probably simplest to use the same exact keyword name.

Also, we would need an entry for this in the docstring.

**Comment by peterm790** [comment]
Thank you @rabernat 
I have committed the recommended changes and updated the docstring!

**Comment by rsignell-usgs** [comment]
@rabernat or @cisaacstern what say ye?

**Comment by cisaacstern** [comment]
Thanks @peterm790 and @rsignell-usgs!

**Pull request merged**